src/core/transpile.jl

```diff
@@ -1417,4 +1417,89 @@ function transpile(::CompressRzGatesTranspiler, circuit::QuantumCircuit)::Quantu
     end
     
     return find_and_compress_blocks(circuit,is_multi_target_or_not_rz,compress_to_rz)
-end
\ No newline at end of file
+end
+
+struct RemoveSwapBySwappingGates <: Transpiler end
+
+"""
+    transpile(::RemoveSwapBySwappingGates, circuit::QuantumCircuit)::QuantumCircuit
+
+Removes the `Swap` gates from the `circuit` assuming all-to-all connectivity.
+
+!!! warning "The initial state must be the ground state!"
+    This transpiler stage assumes that the input state is ``|0\\rangle^{\\otimes N}``
+    where ``N`` is the number of qubits. The stage should not be used on sub-circuits
+    where the input state is not ``|0\\rangle^{\\otimes N}``.
+
+This transpiler stage eliminates `Swap` gates by moving the gates preceding each `Swap`
+gate.
+
+# Examples
+```jldoctest
+julia> transpiler = RemoveSwapBySwappingGates();
+
+julia> circuit = QuantumCircuit(qubit_count=2, gates=[hadamard(1), swap(1,2), sigma_x(2)])
+Quantum Circuit Object:
+   qubit_count: 2 
+q[1]:──H────☒───────
+            |       
+q[2]:───────☒────X──
+                    
+
+
+
+julia> transpiled_circuit = transpile(transpiler, circuit)
+Quantum Circuit Object:
+   qubit_count: 2 
+q[1]:──────────
+               
+q[2]:──H────X──
+               
+
+
+
+```
+"""
+function transpile(::RemoveSwapBySwappingGates, circuit::QuantumCircuit)::QuantumCircuit
+    gates = get_circuit_gates(circuit)
+    qubit_count = get_num_qubits(circuit)
+    output_circuit = QuantumCircuit(qubit_count=qubit_count)
+    qubit_mapping = Dict{Int,Int}()
+    reverse_transpiled_gates = AbstractGate[]
+
+    for gate in reverse(gates)
+        if gate isa Swap
+            update_qubit_mapping!(qubit_mapping, gate)
+        else
+            moved_gate = move_gate(gate, qubit_mapping)
+            push!(reverse_transpiled_gates, moved_gate)
+        end
+    end
+    push!(output_circuit, reverse(reverse_transpiled_gates))
+    return output_circuit
+end
+
+function update_qubit_mapping!(qubit_mapping::Dict{Int,Int}, gate::Swap)
+    connected_qubits = get_connected_qubits(gate)
+    if haskey(qubit_mapping, connected_qubits[1])
+        outlet_qubit_2 = qubit_mapping[connected_qubits[1]]
+        pop!(qubit_mapping, connected_qubits[1])
+        add_first_swap_branch_mapping!(qubit_mapping, connected_qubits)
+        qubit_mapping[connected_qubits[2]] = outlet_qubit_2
+    else
+        add_first_swap_branch_mapping!(qubit_mapping, connected_qubits)
+        qubit_mapping[connected_qubits[2]] = connected_qubits[1]
+    end
```

Comment:
Does it make sense to move `add_first_swap_branch_mapping` to after the if and inline it?

```julia
if haskey(qubit_mapping, connected_qubits[1])
    # ...
else
   # ...
end

if haskey(qubit_mapping, connected_qubits[2])
    # ...
else
   # ...
end
```

Reply:
The `update_qubit_mapping!` function has been restructured as you suggested.